superchain/superchain.go

```diff
@@ -60,6 +60,31 @@ type ChainConfig struct {
 	// Chain is a simple string to identify the chain, within its superchain context.
 	// This matches the resource filename, it is not encoded in the config file itself.
 	Chain string `yaml:"-"`
+
+	// Hardfork Configuration Overrides
+	CanyonTime  *uint64 `yaml:"canyon_time,omitempty"`
+	DeltaTime   *uint64 `yaml:"delta_time,omitempty"`
+	EcotoneTime *uint64 `yaml:"ecotone_time,omitempty"`
+	FjordTime   *uint64 `yaml:"fjord_time,omitempty"`
+}
+
+// replaceMissingOverridesWithDefaults overwrites each unspecified hardfork activation time override
+// with the superchain default.
+func (c *ChainConfig) replaceMissingOverridesWithDefaults(s Superchain) {
+
+	if c.CanyonTime == nil {
+		c.CanyonTime = s.Config.canyonTime
+	}
+	if c.DeltaTime == nil {
+		c.DeltaTime = s.Config.deltaTime
+	}
+	if c.EcotoneTime == nil {
+		c.EcotoneTime = s.Config.ecotoneTime
+	}
+	if c.FjordTime == nil {
+		c.FjordTime = s.Config.fjordTime
+	}
```

Comment:
It is easy to miss adding new fields here. Could we add a test that fails if we add a field to the superchain config, but not to the chain config, and if present in both checks that these overrides are applied?

Reply:
Don't we want to allow chains to omit fields in the chain config, though? If we do not, then there's no point in having the fields in the superchain config and we can simplify things a lot.

Reply:
Hesitate to suggest it, but we could roll something with reflection to ensure we didn't forget to add all necessary fields here.